python/sdk/test/integration_test.py

```diff
@@ -1034,3 +1013,131 @@ def test_custom_model_with_artifact(
 
     # Undeploy other running model version endpoints
     undeploy_all_version()
+
+
+@pytest.mark.integration
+@pytest.mark.dependency()
+def test_deployment_mode(integration_test_url, project_name, use_google_oauth):
+    """
+    Validate that user can redeploy a model version using different deployment mode
+    """
+    merlin.set_url(integration_test_url, use_google_oauth=use_google_oauth)
+    merlin.set_project(project_name)
+    merlin.set_model("deployment-mode", ModelType.TENSORFLOW)
+    model_dir = "test/tensorflow-model"
+
+    undeploy_all_version()
+
+    with merlin.new_model_version() as v:
+        merlin.log_model(model_dir=model_dir)
+
+    # Deploy using serverless
+    initial_endpoint = merlin.deploy(v)
+
+    resp = requests.post(f"{initial_endpoint.url}", json=tensorflow_request_json)
+
+    assert resp.status_code == 200
+    assert resp.json() is not None
+    assert len(resp.json()["predictions"]) == len(tensorflow_request_json["instances"])
+
+    # Deploy using raw_deployment
+    new_endpoint = merlin.deploy(v, deployment_mode=DeploymentMode.RAW_DEPLOYMENT)
+
+    assert new_endpoint.url == initial_endpoint.url
+    resp = requests.post(f"{new_endpoint.url}", json=tensorflow_request_json)
+
+    assert resp.status_code == 200
+    assert resp.json() is not None
+    assert len(resp.json()["predictions"]) == len(tensorflow_request_json["instances"])
+
+    # Deploy again using serverless with concurrency autoscaling policy
+    new_endpoint = merlin.deploy(v, deployment_mode=DeploymentMode.SERVERLESS,
+                                 autoscaling_policy=merlin.AutoscalingPolicy(
+                                     metrics_type=merlin.MetricsType.CONCURRENCY,
+                                     target_value=2))
+
+    assert new_endpoint.url == initial_endpoint.url
+    resp = requests.post(f"{new_endpoint.url}", json=tensorflow_request_json)
+
+    assert resp.status_code == 200
+    assert resp.json() is not None
+    assert len(resp.json()["predictions"]) == len(tensorflow_request_json["instances"])
+
+    merlin.undeploy(v)
+
+
+@pytest.mark.integration
+@pytest.mark.dependency()
+def test_deployment_mode_for_serving_model(integration_test_url, project_name, use_google_oauth):
```

Comment:
The name suggests the test updates the deployment mode of serving model version, instead, the current test will deploy new model version with different deployment mode and then serve it. Would it be possible to test updating the deployment mode of serving model? E.g., model version 1 with serverless mode is serving, then user want to try to redeploy this model version 1 with raw deployment mode.

Reply:
Updated